### ISSUE TITLE
Compatibility with PHPUnit 7.4

### ISSUE THREAD

**Issue by remicollet** [php-mock/php-mock-phpunit#29]
Running zend-ldap test suite

`Fatal error:  Declaration of phpmock\phpunit\MockObjectProxy::__phpunit_verify() must be compatible with PHPUnit_Framework_MockObject_MockObject::__phpunit_verify(bool $unsetInvocationMocker = true) in ...`

**Comment by sebastianbergmann** [comment]
That is why one should not depend on private implementation details.

**Comment by chrisminett** [comment]
My first thought was that this might be similar to when this project failed to work against the minor update to PHPUnit 6.5 https://github.com/sebastianbergmann/phpunit-mock-objects/issues/390 . In that case however, the underlying phpunit-mock-objects had a major version increase to v5.0.0 and the fix was not so simple: https://github.com/php-mock/php-mock-phpunit/pull/20

This time a public method on the MockObject interface has changed its signature in a minor version of PHPUnit, which does seem like a BC break as any implementation of this method will no longer be compatible.

Fortunately updating the method signature in this project will work with PHPUnit 7.3 and 7.4 as the extra parameter in the implementation does not affect compatibility with the lack of the parameter in PHPUnit 7.3:

```php
interface A {
    public function foo();
}
class B {
    public function foo(bool $newParam = true) {
        echo 'bar';
    }
}
(new B())->foo();
```

**Comment by webimpress** [comment]
Fixed in #30 and released in version [2.1.2](
https://github.com/php-mock/php-mock-phpunit/releases/tag/2.1.2).
